### ISSUE TITLE
React error when using both useForm and a custom hook using useState 

### ISSUE THREAD

**Issue by jfreyheit** [fabian-hiller/modular-forms#250]
The following code crashes with a React error "Uncaught Error: Should have a queue. This is likely a bug in React. Please file an issue." when the button is clicked.

![ShouldHaveAQueue](https://github.com/user-attachments/assets/43f61e95-731d-45ad-bf65-750868bba0a1)

If I remove the useForm call it works properly

```
import { useForm } from "@modular-forms/react";
import { useState } from "react";

type UseItemsOutput<T> = {
  items: T[];
  onAdd: () => void;
};

function useItems<T>(
  defaultItems: T[],
  itemFactory: () => T
): UseItemsOutput<T> {
  const [items, setItems] = useState(defaultItems);

  function onAdd(): void {
    setItems([...items, itemFactory()]);
  }

  return {
    items,
    onAdd,
  };
}

type FormData = { a: string };

function App() {
  const { items, onAdd } = useItems([], () => 1);
  useForm<FormData>({
    initialValues: { a: "" },
  });

  return (
    <div>
      <h1>{items.length}</h1>
      {items.map((item) => (
        <p>{item}</p>
      ))}
      <button onClick={onAdd}>Add</button>
    </div>
  );
}

export default App;
```

Dependencies are very straightforward : 

"@modular-forms/react": "^0.9.0",
    "@preact/signals-react": "^1.3.8",
    "clsx": "^2.1.1",
    "react": "^18.3.1",
    "react-dom": "^18.3.1",
    "valibot": "^0.42.0"

[Stackblitz repro](https://stackblitz.com/edit/vitejs-vite-f3svok)

**Comment by fabian-hiller** [comment]
Can you try to update your packages to the latest version? `@preact/signals-react` seems to be outdated.

**Comment by jfreyheit** [comment]
Hi, I've updated to the latest versions (mostly @preact/signals-react 2.2.0) but I get the same error. I've updated the Stackblitz playground if you want to have a look.

**Comment by fabian-hiller** [comment]
Hmm... have you tried to debug it? It is very likely that this bug is not related to our code, as we just reused the function provided by React and Preact. Feel free to take a look at our playground: https://stackblitz.com/edit/modular-forms-preact?file=src/routes/login.tsx

**Comment by jfreyheit** [comment]
I have absolutely no idea how to debug that kind of stuff. However I still gave it some thought and after toying around, I realized that the example worked if I uncommented @preact/signals-react's auto-registration (import "@preact/signals-react/auto";).

I managed to use their babel plugin instead and there's no more crash. It may be worth mentioning this in the documentation so that people don't use the auto-registration.
